modules/zip_importer/includes/importer.inc

```diff
@@ -178,10 +178,29 @@ class ZipBatchImporter extends IslandoraBatchImporter {
 class ZipBatchImportObject extends IslandoraImportObject {
 
   protected $mods;
-  protected static $MARC2MODS;
```

Comment:
Sorry, as i am new to this project i just followed whatever was done there and fixed the issue which we were facing. Ideally these should be configured via admin interface and it should be selectable at the time of importing - which will give the flexibility to use the custom xsl transformations as well.
So like you said it should be implemented via the drupal hook system.